aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -230,131 +414,133 @@ For more information, see [Scope validation in the Web Host topic](xref:fundamen
 
-### Disposing of services
+### Disposal of services
 
-The container will call `Dispose` for `IDisposable` types it creates. However, if you add an instance to the container yourself, it will not be disposed.
-
-Example:
+The container calls `Dispose` for the `IDisposable` types it creates. If an instance is added to the container by user code, it isn't disposed automatically.
 
 ```csharp
-// Services implement IDisposable:
+// Services that implement IDisposable:
 public class Service1 : IDisposable {}
 public class Service2 : IDisposable {}
 public class Service3 : IDisposable {}
 
 public interface ISomeService {}
 public class SomeServiceImplementation : ISomeService, IDisposable {}
 
-
 public void ConfigureServices(IServiceCollection services)
 {
-    // container will create the instance(s) of these types and will dispose them
+    // The container creates the following instances and disposes them automatically:
     services.AddScoped<Service1>();
     services.AddSingleton<Service2>();
     services.AddSingleton<ISomeService>(sp => new SomeServiceImplementation());
 
-    // container didn't create instance so it will NOT dispose it
+    // The container doesn't create the following instances, so it doesn't dispose of
+    // the instances automatically:
     services.AddSingleton<Service3>(new Service3());
     services.AddSingleton(new Service3());
 }
 ```
 
+::: moniker range="= aspnetcore-1.0"
+
 > [!NOTE]
-> In version 1.0, the container called dispose on *all* `IDisposable` objects, including those it didn't create.
+> In ASP.NET Core 1.0, the container calls dispose on *all* `IDisposable` objects, including those it didn't create.
 
-## Replacing the default services container
+::: moniker-end
 
-The built-in services container is meant to serve the basic needs of the framework and most consumer applications built on it. However, developers can replace the built-in container with their preferred container. The `ConfigureServices` method typically returns `void`, but if its signature is changed to return `IServiceProvider`, a different container can be configured and returned. There are many IOC containers available for .NET. In this example, the [Autofac](https://autofac.org/) package is used.
+## Default service container replacement
 
-First, install the appropriate container package(s):
+The built-in service container is meant to serve the basic needs of the framework and most consumer apps built on it. However, developers can replace the built-in container with their preferred container. The `Startup.ConfigureServices` method typically returns `void`. If the method's signature is changed to return an [IServiceProvider](/dotnet/api/system.iserviceprovider), a different container can be configured and returned. There are many IoC containers available for .NET. In the following example, the [Autofac](https://autofac.org/) container is used:
```

Comment:
Confirm with engineering, but I seem to remember something related to this wasn't possible until ASP.NET Core 1.1. As noted in the Autofac docs, "ASP.NET Core 1.1 introduced the ability to have strongly typed container configuration." Also, I have a personal 2.0 app using Autofac with a return type of `void` in `Startup.ConfigureServices`.

Reply:
ok ... will do.

Reply:
`ConfigureContainer` looks like it was 1.1, but returning `IServiceProvider` from `ConfigureServices` was 1.0, so I think we're ok as far as this passage is concerned. https://github.com/aspnet/Hosting/issues/91